File: 39/p39.py
from math import sqrt
p=[]
for a in range(1,999):
  for b in range(1,min(a,1001-2*a)):
    c=sqrt(a*a+b*b)
    if c%1==0:
      p.append((a,b,c,a+b+c))
p.sort(key=lambda t:t[3])
v=p[0][3]
l=0
m=0
n=0
for t in p:
  if t[3]!=v:
    if l>m:
      m=l
      n=v
    l=0
    v=t[3]
  l+=1
print(n)
  

File: 39/p39.cxx
#include <math.h>
#include <stdio.h>
int main(){
  int p[2000];
  for(int i=0;i<2000;++i)p[i]=0;
  int c,m=0,n=0;
  for(int a=1;a<999;++a){
    for(int b=1;(b<=a)&&(b<1001-2*a);++b){
      if((c=sqrt(double(a*a+b*b)))%1==0)++p[a+b+c];
    }
  }
  for(int i=0;i<1001;++i){
    if(p[i]>m){
      m=p[i];
      n=i;
    }
  }
  printf("%i %i\n",n,m);
}

